addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SinkDetectionCollectAndRefreshParamValues.java

```diff
@@ -64,17 +67,55 @@ public String getReference() {
     }
 
     @Override
-    public void scan(HttpMessage msg, String param, String value) {
+    public void scan() {
+        // refresh the state of the website
+        HttpMessage msg = getBaseMsg();
         try {
             HttpMessage msg1 = msg.cloneRequest();
-            this.setParameter(msg1, param, PersistentXssUtils.getUniqueValue(msg1, param));
-            if (log.isDebugEnabled()) {
-                log.debug("Prime msg=" + msg1.getRequestHeader().getURI() + " param=" + param);
-            }
             sendAndReceive(msg1, false);
         } catch (Exception e) {
             log.error(e.getMessage(), e);
         }
+        super.scan();
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        // TODO: should collect params only on POST/PUT when low or medium strength?
```

Comment:
may be `path param`

Reply:
Can you explain a bit more? I don't understand.

Reply:
My mistake i thought you have written a comment, but it is a TODO. Ignore